tests/test_api.py

```diff
@@ -190,3 +195,66 @@ def test_is_enabled(self):
         assert not api.is_enabled_for_course(course_id)
         api.set_dates_for_course(course_id, items)
         assert api.is_enabled_for_course(course_id)
+
+    def test_allow_relative_dates(self):
+        course_key = CourseLocator('testX', 'tt101', '2019')
+        block1 = make_block_id(course_key)
+        date1 = datetime(2019, 3, 22)
+        block2 = make_block_id(course_key)
+        date2 = datetime(2019, 3, 23)
+        date2_override_delta = timedelta(days=10)
+        date2_override = self.schedule.start_date + date2_override_delta
+        block3 = make_block_id(course_key)
+        date3_delta = timedelta(days=1)
+        date3 = self.schedule.start_date + date3_delta
+        block4 = make_block_id(course_key)
+        date4_delta = timedelta(days=2)
+        date4 = self.schedule.start_date + date4_delta
+        date4_override = datetime(2019, 4, 24)
+        items = [
+            (block1, {'due': date1}),  # absolute
+            (block2, {'due': date2}),  # absolute, to be overwritten by relative date
+            (block3, {'due': date3_delta}),  # relative
+            (block4, {'due': date4_delta}),  # relative, to be overwritten by absolute date
+        ]
+        api.set_dates_for_course(course_key, items)
+        api.set_date_for_block(course_key, block2, 'due', date2_override_delta, user=self.user)
+        api.set_date_for_block(course_key, block4, 'due', date4_override, user=self.user)
+
+        # get_dates_for_course
+        dates = [
+            ((block1, 'due'), date1),
+            ((block2, 'due'), date2),
+            ((block3, 'due'), date3),
+            ((block4, 'due'), date4),
+        ]
+        assert api.get_dates_for_course(course_key, schedule=self.schedule) == dict(dates)
```

Comment:
Should we be trying to move towards `self.assertEqual`? I can't remember if there was one that was preferred by edX

Reply:
I dunno? `edx-when` certainly seems to prefer bare asserts. And pytest's documentation only shows base asserts: https://docs.pytest.org/en/latest/assert.html

However... even that documentation says that the nice assert formatting doesn't work if you provide an assert message. And some things need the method approach (assertContains which does a few various http request checks or assertRaises, and probably others). So in my own head, I kind of prefer the method style. But I've seen hints that true Pythonistas should prefer bare asserts...

Reply:
IIRC the original version of edx-when did not have its test classes inheriting from anything which might be why everything is written in bare asserts rather than methods. So I think it's safe to update if we have a preference. I've no strong feelings either way, but might be good to document somewhere so we don't end up in the case where we have 50% each way.